### ISSUE TITLE
Values cached using play.api.cache.Cached never expire

### ISSUE THREAD

**Issue by gabfssilva** [playframework/playframework#9714]
### Play Version

2.7.3

### API

Scala

### Operating System

Ubuntu

### JDK

OpenJDK 11 - OpenJ9

### Library Dependencies

cache, play-redis

### Expected Behavior

Using Cached, I want to set a timeout for a value to expire, using default.

So, after 10 seconds the value expires from the cache so the action is executed once again.

### Actual Behavior

The value never expires

### Reproducible Test Case

Create an action using cached and default:

```scala
cached(s"fetchById#$id").default(10.seconds) {
  Action.async { 
    //result here
  }
}
```

The action is called once and then never called again.

### The issue

`default` implementation

```scala
def default(duration: Duration): CachedBuilder = compose({ case _: ResponseHeader => duration })

def compose(alternative: PartialFunction[ResponseHeader, Duration]): CachedBuilder = new CachedBuilder(
    cache = cache,
    key = key,
    caching = caching.orElse(alternative)
  )
```

basically, this line `caching = caching.orElse(alternative)` means that: if there's a caching defined before, ignores the alternative (which is the default duration value).

The problem is that cached.apply(key: String) already sets a caching value, so, anything after is ignored.

The workaround:

```scala
cached((_: RequestHeader) => s"fetchById#$id", 10.seconds) {
  Action.async { 
    //result here
  }
}
```

Is this really a bug or I just didn't understand the proposed behavior?
